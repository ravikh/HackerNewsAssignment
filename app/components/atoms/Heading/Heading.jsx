import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import styles from './Heading.style';

const Heading = ({ type, children, appear, className }) => React.createElement(styled(type)`${styles}`, {type, appear, className}, children)

Heading.defaultProps = {
    type: 'h3',
    appear: '',
    className: ''
};

Heading.propTypes = {
    type: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),
    appear: PropTypes.string,
    className: PropTypes.string
};

export default Heading;