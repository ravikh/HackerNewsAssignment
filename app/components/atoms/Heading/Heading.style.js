
import {css} from 'styled-components';

const styleRules = (props) => {
    let cssRules = props.theme.typography.font_default;
    if (props.type) {
        if (props.appear) {
            cssRules = props.theme.typography[`font_${props.appear}`]
        } else {
            cssRules = props.theme.typography[`font_${props.type}`]
        }
    }
    return cssRules;
}

const HeadingStyles = css`
    ${props => styleRules(props)}
`;

export default HeadingStyles;