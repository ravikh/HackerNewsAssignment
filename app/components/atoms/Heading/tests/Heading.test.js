import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Heading from '../Heading';

import Theme from '../../../../theme';

describe('Heading', () => {
  let heading = '';
  const theme = { ...Theme };

  test('should render correctly', () => {
    heading = shallow(<Heading theme={theme} type="h1"/>);
    expect(heading).toMatchSnapshot();
  });
});