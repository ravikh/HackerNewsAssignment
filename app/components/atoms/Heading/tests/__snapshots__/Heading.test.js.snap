// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render correctly 1`] = `
<styled.h1
  appear=""
  type="h1"
/>
`;
